maestro-android/src/main/java/dev/mobile/maestro/MainActivity.kt

```diff
@@ -0,0 +1,41 @@
+package dev.mobile.maestro
+
+import android.os.Bundle
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.IntentFilter
+import android.util.Log
+import dev.mobile.maestro.receivers.HasAction
+import dev.mobile.maestro.receivers.LocaleSettingReceiver
+
+class MainActivity : Activity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.main)
+
+        Log.d(TAG, "Entering the app")
+
+        registerSettingsReceivers(listOf(
+            LocaleSettingReceiver::class.java
+        ))
```

Comment:
We wouldn't need to register receiver here. Mentioning in manifest should suffice you can issue adb broadcast to cross check.